cirq/contrib/qcircuit/optimizers.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cirq import circuits, ops
+
+class _PadAfterSwapGates(circuits.OptimizationPass):
```

Comment:
I'm not sure I like this optimization pass. Is there some other way to do this that's more coupled to the method that produces the qcircuit diagram, or to the swaps themselves?

Reply:
Yes, I feel like if we kept implementing features this way we'd have a giant mess in no time. We need something simpler that includes this use case as a sub-case.

This feels like code that a user would add on to make the library work the way they like, and it would do well at that. But in the library itself it seems like a piece that doesn't play well with the other pieces.

Reply:
This is exactly code that a user (me) added on to make the library work the way I wanted. I care less that it be in the library because it's easy enough to tack on afterwards (so long as `circuit_to_latex_using_qcircuit` takes an `optimizers` argument). Same goes for everything in this PR except `TextDiagramDrawer.horizontal_occlusions` and `CircuitDiagramInfo.hconnected`.